### ISSUE TITLE
[adapter-static] router should work for .html files

### ISSUE THREAD

**Issue by omar2205** [sveltejs/kit#4794]
### Describe the problem

When deploying a static site and trying to visit the individual pages (not from the index page) 
the router doesn't serve the HTML files.

/test1.svelte -> test1.html -> Not found: /test1.html
/test2.html.svelte -> test2.html -> Works

Repo: https://github.com/omar2205/sk-static-route-html

### Describe the proposed solution

It should serve the HTML file

### Importance

would make my life easier

**Comment by tcc-sejohnson** [comment]
Okay, a few notes. A maintainer can come in behind me and correct anything that's wrong.

When naming your route files, you must add `.html.svelte` if you want the _route_ to that file to be `.html.svelte`. This is intended behavior:

`{fileName} => {resultingRoute}`
`test1.svelte => /test1`
`test2.html.svelte => /test2.html`
`test3.html.grandpa.svelte => /test2.html.grandpa`

When running your repo using `npm run preview`, everything works as expected. Visiting `/test1` renders the correct page; visiting `/test2.html` also does (try it!). I believe there's a configuration issue with your Python HTTP server. When serving the site from Python, visiting `localhost:8000` allows Kit to initialize its router, meaning the client-side links you click all work. When trying to visit `/test1`, the Python server tries to look for the literal file `/test1`, which doesn't exist. When trying to visit `/test1.html`, the Python server tries to look for that file and finds it, which allows it to serve a webpage -- except, now, you're at `/test1.html`, which, to your Svelte app, is not a valid route.

TL;DR, you should _not_ be trying to serve the actual files by name from your `build` folder -- that's not how it's supposed to work.

As for what the correct configuration for your server is, I'm not sure.

**Comment by stephane-vanraes** [comment]
In the reproduction you posted the build is not creating any files because you are missing one of two options (which one depends on what you are trying to achieve) but in neither cases is there a need to name your files `.html.svelte` that that works is just a coincidence due to some internal workings.

## Prerender all pages

This applies if you want all pages to be fully prerendered, meaning **all** visitors will see the same result:

```js
const config = {
  kit: {
    adapter: adapter(),		
    prerender: { default: true }
  }
};
```

## Use a fallback page

This is useful if you want to create an SPA where prerendering wouldn't work because the content depends on the user (or is generally dynamic)

```js
const config = {
   kit: {
     adapter: adapter({
	fallback: 'index.html'
    }),
};
```

---

You can also combine both systems to make a sort of 'hybrid', marking individual pages to not be prerendered.
```html
<script context="module">
    export const prerender = false;
</script>
```

Currently the documentation for adapters and some of the usage is not 100%, but the current focus is more on getting SvelteKit to version 1.0 and a touch less on updating documentation for adapters that might still change slightly.

**Comment by omar2205** [comment]
My intention was in cases where I can't have a fallback, for example, a simple server like Python's. My thinking was that if the pages are compiled to `.html` anyway, the router should just render them. It's a low priority issue behind all the 1.0 milestones.

**Comment by antony** [comment]
It seems that there is a problem with the server you are deploying to. Accessing /test2.html (or any other html page generated by the static adapter) works without issue.

You can verify this by creating the demo app, installing + importing the static adapter, and then running:

```
npm run build
npx http-server build
```

Then visit http://localhost:8080/about.html
